### PR TITLE
Input file support for `Resize` move.

### PR DESCRIPTION
**Pull request by g-bauer**
This PR does the following:

- Implements input file support for `Resize` move.
- Adds input tests.
- Adds documentation in the user manual.
- More detailed explanations for the whole `Monte-Carlo` chapter in the manual.

I overhauled the `Monte-Carlo` chapter in the manual and I'd like to hear your feedback on that part specifically. Is it written in a too informal style? Do you like the presentation? Is there too much text (maybe users just want to know what keywords they need)? I'd like to do the same with the MD chapter.

### CODE REVIEW

File: doc/src/input/mc.md [hunk 1]
```diff
@@ -1,8 +1,27 @@
...
 
-A Monte-Carlo simulation is started by setting the propagator `type` to
-`MonteCarlo`. The two needed keys are the `temperature` of the simulation, and
-the Monte-Carlo `moves` to use.
+If you want to perform a Monte-Carlo simulation, you have to set the propagator 
+`type` to `"MonteCarlo"`.
+Every Monte-Carlo simulations needs a `temperature` and a set of `moves` 
+(this set can consist of a single move).
+
+You can think of a "move" as a specific instruction to change the system.
```
Comment (Luthaf) [inline doc/src/input/mc.md]: "change the system" => "generate a new test configuration"?

File: doc/src/input/mc.md [hunk 2]
```diff
@@ -11,64 +30,121 @@ temperature = "500 K"
...
+
+#### Example
 
 ```toml
 [simulations.propagator]
 type = "MonteCarlo"
 temperature = "500 K"
 moves = [
     {type = "Translate", delta = "1 A", frequency = 2},
+    # for He, use a larger maximum displacement.
```
Comment (Luthaf) [inline doc/src/input/mc.md]: Here, "He" would be translated by both the main move and the second move. Maybe specify/explicit it somewhere?

File: src/input/tests/simulation/bad/mc/move-14.toml [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+[[simulations]]
+nsteps = 1
+
+[simulations.propagator]
+type = "MonteCarlo"
+temperature = "300 K"
+moves = [
+    {type = "Resize", pressure = "5.0 bar", delta = "6 A^3", frequency = "3"}
+    #^ 'frequency' must be a number in Monte-Carlo move
+]
```
Comment (Luthaf) [inline src/input/tests/simulation/bad/mc/move-14.toml]: Can you add a new line at the end of the file?

File: src/input/tests/simulation/bad/mc/move-15.toml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+[[simulations]]
+nsteps = 1
+
+[simulations.propagator]
+type = "MonteCarlo"
+temperature = "300 K"
+moves = [
+    {type = "Resize", delta = "6 A^3"}
+    #^ Missing 'pressure' key in Resize move
+]
```
Comment (Luthaf) [inline src/input/tests/simulation/bad/mc/move-15.toml]: Can you add a new line at the end of the file?

File: src/input/tests/simulation/bad/mc/move-16.toml [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+[[simulations]]
+nsteps = 1
+
+[simulations.propagator]
+type = "MonteCarlo"
+temperature = "300 K"
+moves = [
+    {type = "Resize", delta = "6 A^3", pressure = 6}
+    #^ 'pressure' must be a string in Resize move
+]
```
Comment (Luthaf) [inline src/input/tests/simulation/bad/mc/move-16.toml]: Can you add a new line at the end of the file?

**Comment by Luthaf** [comment]
> I overhauled the Monte-Carlo chapter in the manual and I'd like to hear your feedback on that part specifically. Is it written in a too informal style? Do you like the presentation? Is there too much text (maybe users just want to know what keywords they need)? I'd like to do the same with the MD chapter.

I like this, it is better, and not too verbose.

> (maybe users just want to know what keywords they need)?

There is a bit of a tension in the current user manual, cause it tries to be at the same time an user manual, and a reference for the input. We may want to separated both parts, but I don't know how it could fit together.

**Comment by g-bauer** [comment]
Thanks for the feedback.

> There is a bit of a tension in the current user manual, cause it tries to be at the same time an user manual, and a reference for the input.

To reduce this tension, I put the links to all moves at the top of the page. So if one wants to take a quick look for input parameters one can navigate to the specific move fastly. But you are right, at the moment it is a compromise between input reference and manual. I think for now that is fine.

File: doc/src/input/mc.md (hunk 2 above)
Comment (g-bauer) [inline doc/src/input/mc.md]: I added more text to make this explicit.

File: doc/src/input/mc.md (hunk 1 above)
Comment (g-bauer) [inline doc/src/input/mc.md]: Thanks! I took your suggestion and made this more explicit.

**Comment by g-bauer** [comment]
Added blank lines at the end of all test files in `/lumol/src/input/tests/simulation/bad/mc`.

**Pull request merged**